frontend/src/lib/v2/WorkflowUtils.ts

```diff
@@ -111,12 +117,19 @@ export function isPipelineSpec(templateString: string) {
 export function getContainer(componentSpec: ComponentSpec, templateString: string) {
   const executionLabel = componentSpec?.executorLabel;
 
-  const jsonTemplate = getPipelineDefFromYaml(templateString);
-  const deploymentSpec = jsonTemplate['deploymentSpec'];
+  const pipelineSpecYAML = getPipelineDefFromYaml(templateString);
+  const pipelineSpec = PipelineSpec.fromJSON(pipelineSpecYAML);
+  const deploymentSpecObj = pipelineSpec.deploymentSpec;
+  if (!pipelineSpec || !deploymentSpecObj) {
```

Comment:
If `pipelineSpec` is already checked like my other comment above, I guess we only need to check for `deploymentSpecObj` here?

Reply:
This comment seems not addressed yet.